www/class/config-generate/abstract/JsonFormat.class.php

```
@@ -0,0 +1,88 @@
+ * permission to link this program with independent modules to produce an executable,
+ * regardless of the license terms of these independent modules, and to copy and
+ * distribute the resulting executable under terms of Centreon choice, provided that
+ * Centreon also meet, for each linked independent module, the terms  and conditions
+ * of the license of that module. An independent module is a module which is not
+ * derived from this program. If you modify this program, you may extend this
+ * exception to your version of the program, but you are not obliged to do so. If you
+ * do not wish to do so, delete this exception statement from your version.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+abstract class JsonFormat
+{
+    /**
+     * @var mixed
+     */
+    protected mixed $cacheData = null;
+
+    /**
+     * @var string|null
+     */
+    protected ?string $filePath = null;
+
+    /**
+     * @param mixed $data
+     */
+    public function setContent(mixed $data): void
+    {
+        $this->cacheData = $data;
+    }
+
+    /**
+     * Defines the path of the file where the data should be written.
+     *
+     * @param string $filePath
+     */
+    public function setFilePath(string $filePath): void
+    {
+        $this->filePath = $filePath;
+    }
+
+    /**
+     * Writes the content of the cache only if it is not empty.
+     *
+     * @return int|bool Number of bytes written or false on failure
+     * @throws Exception
+     */
+    public function flushContent(): int|bool
+    {
+        if ($this->filePath === null) {
+            throw new Exception('No file path defined');
+        }
+        if (! empty($this->cacheData)) {
+            $data = json_encode($this->cacheData, JSON_UNESCAPED_SLASHES | JSON_PRETTY_PRINT);
+            $this->cacheData = null;
+
+            return file_put_contents($this->filePath, $data);
```

Could you throw an exception on failure